### ISSUE TITLE
Why require Internet? 

### ISSUE THREAD

**Issue by flickerfly** [RedHatOfficial/ocp4-helpernode#84]
I'd be interested in using this with a offline install. I see the vars.yaml with URLs in it. Could I simply put those in my own location and remove that dependency or are there other reasons I need this? 

If so, I'm wondering if a script to download those, drop them in files/ and change the variable to point to that would be enough to drop the dependency. I'd just copy the result into my Ansible code.

**Comment by christianh814** [comment]
> I'd be interested in using this with a offline install. 

You definitely  can as long as you [follow the disconnected install docs](https://docs.openshift.com/container-platform/4.4/installing/installing_bare_metal/installing-restricted-networks-bare-metal.html) ...you may also need to [sync the repos](https://linux.die.net/man/1/reposync) and [create your own internal repo](https://linux.die.net/man/8/createrepo) in order to install the helpernode.

FWIW, I've done a disconnected install with the helpernode lots of times.

> I see the vars.yaml with URLs in it. Could I simply put those in my own location and remove that dependency or are there other reasons I need this?

So in your `vars.yaml` file you can do something like this

```
ocp_bios: "file:///path/to/rhcos-4.4.3-x86_64-metal.x86_64.raw.gz"
ocp_initramfs: "file:///path/to/rhcos-4.4.3-x86_64-installer-initramfs.x86_64.img"
ocp_install_kernel: "file:///path/to/rhcos-4.4.3-x86_64-installer-kernel-x86_64"
ocp_client: "file:///path/to/openshift-client-linux-4.4.3.tar.gz"
ocp_installer: "file:///path/to/openshift-install-linux-4.4.3.tar.gz"
```

And just copy these files over via "sneaker net" into the helper


> If so, I'm wondering if a script to download those, drop them in files/ and change the variable to point to that would be enough to drop the dependency. I'd just copy the result into my Ansible code.

We've discussed including the binaries with the playbook (@therevoman , @vchintal, and myself) and we decided against it. Copying the binaries effectively ties the helper to a version of OpenShift. Which, in my mind, isn't the point of this repo.

Changing the `vars.yaml` provides this functionality anyway, and doesn't tie this playbook to a specific version of OpenShift.

HTH

**Comment by flickerfly** [comment]
Thanks, that's some good information. 

I agree that the images you link to don't belong here, but are better as a dependency. After seeing your demo with Chris Short, I realized that the script I'm theorizing is just basically downloading them and setting the variables in vars.yaml I guess. (Tons of good info in that demo! Thanks!)

As I'd need to sneakernet my data (not have a dual-homed bastion), seems like it wouldn't be hard to host that rpm repo and container repo on the helpernode since it already has a web server. I'd have to have the same repos setup elsewhere and then copy over the underlying volumes I guess,

**Comment by christianh814** [comment]
> Thanks, that's some good information.
> 
> I agree that the images you link to don't belong here, but are better as a dependency. After seeing your demo with Chris Short, I realized that the script I'm theorizing is just basically downloading them and setting the variables in vars.yaml I guess. (Tons of good info in that demo! Thanks!)
> 
> As I'd need to sneakernet my data (not have a dual-homed bastion), seems like it wouldn't be hard to host that rpm repo and container repo on the helpernode since it already has a web server. I'd have to have the same repos setup elsewhere and then copy over the underlying volumes I guess,

You def can. Also, there is a plan to [include a registry with the helper](https://github.com/RedHatOfficial/ocp4-helpernode/issues/9) in the future. If you get this working, we'd love a PR :smile:

**Comment by flickerfly** [comment]
Already subscribed to it. Just in the planning stages, but If I get to it before anyone else, will do.

**Comment by christianh814** [comment]
Closed by #93
